isthmus/src/test/java/io/substrait/isthmus/CalciteLiteralTest.java

```diff
@@ -194,6 +194,29 @@ void tIntervalYearMonthWithPrecision() {
         convertedRex.getValueAs(BigDecimal.class).longValue());
   }
 
+  @Test
+  void tIntervalDaySecond() {
+    // Calcite stores milliseconds since Epoch, so test only millisecond precision
+    BigDecimal bd =
+        new BigDecimal(
+            TimeUnit.DAYS.toMillis(3)
+                + TimeUnit.HOURS.toMillis(5)
+                + TimeUnit.MINUTES.toMillis(7)
+                + TimeUnit.SECONDS.toMillis(9)
+                + 500); // '3-5:7:9.500' day to second (6)
+    RexLiteral intervalDaySecond =
+        rex.makeIntervalLiteral(
+            bd,
+            new SqlIntervalQualifier(
+                org.apache.calcite.avatica.util.TimeUnit.DAY,
+                -1,
+                org.apache.calcite.avatica.util.TimeUnit.SECOND,
+                3,
+                SqlParserPos.QUOTED_ZERO));
```

Comment:
I was taking a look and noticed that we've defined the same intervals in 3 places effectively:
1. https://github.com/substrait-io/substrait-java/blob/1a4b7b1d729f87b9ccd310aef976568f3d867590/isthmus/src/main/java/io/substrait/isthmus/TypeConverter.java#L45-L48
2. https://github.com/substrait-io/substrait-java/blob/1a4b7b1d729f87b9ccd310aef976568f3d867590/isthmus/src/main/java/io/substrait/isthmus/expression/ExpressionRexConverter.java#L42-L48
3. Now in these tests.

I think it would be good to define them in one place, make them public and re-use them. I think `SubstraitTypeSystem` would be a good home for them.

Reply:
I factorized a bit, but since the tests do tweak precision they still create some specific SqlIntervalQualifier objects.

Reply:
Factorizing out only the really common ones makes sense.